beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -2370,9 +2376,9 @@ mod tests {
                     outgoing: bool::arbitrary(g),
                     attestation_net_bitfield,
                     sync_committee_net_bitfield,
-                    score: f64::arbitrary(g),
+                    score: rand_f64(),
```

Comment:
Quickcheck (and proptest) can do a clever thing called _shrinking_ where they simplify (shrink) failing test cases to be as small as possible. If we mix in `rand` we are taking control from quickcheck which will make the test non-deterministic based on quickcheck's seed _and_ will disable shrinking.

I think it might be better to use `f64::arbitrary` in a loop until it returns a value other than `NaN`. An even cleaner option might be generating floats in a range, which would also avoid other weird values like infinity. From a quick skim of the docs I'm not sure if quickcheck's arbitrary has a function for this though

Reply:
We could do the `NaN` loop for now and then come back to this if/when we rewrite our quickcheck tests using proptest. Proptest is a bit better maintained

Reply:
thanks for the additional context!

Reply:
aa9014a